src/sagas/walletConnect.js

```diff
@@ -0,0 +1,424 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { Core } from '@walletconnect/core';
+import { Web3Wallet } from '@walletconnect/web3wallet';
+
+import {
+  call,
+  fork,
+  take,
+  all,
+  put,
+  cancel,
+  cancelled,
+  takeLatest,
+  takeEvery,
+  select,
+  race,
+} from 'redux-saga/effects';
+import { eventChannel } from 'redux-saga';
+import { get } from 'lodash';
+
+import { WalletConnectModalTypes } from '../components/WalletConnect/WalletConnectModal';
+import {
+  WALLET_CONNECT_PROJECT_ID,
+  WALLET_CONNECT_FEATURE_TOGGLE,
+  NETWORK,
+} from '../constants';
+import {
+  types,
+  setWalletConnect,
+  setWalletConnectModal,
+  setWalletConnectSessions,
+} from '../actions';
+import { checkForFeatureFlag } from './helpers';
+
+const AVAILABLE_METHODS = ['hathor_signMessage'];
+const AVAILABLE_EVENTS = [];
+
+/**
+ * Those are the only ones we are currently using, extracted from
+ * https://docs.walletconnect.com/2.0/specs/clients/sign/error-codes
+ */
+const ERROR_CODES = {
+  UNAUTHORIZED_METHODS: 3001,
+  USER_DISCONNECTED: 6000,
+  USER_REJECTED: 5000,
+  USER_REJECTED_METHOD: 5002,
+};
+
+function* isWalletConnectEnabled() {
+  const walletConnectEnabled = yield call(checkForFeatureFlag, WALLET_CONNECT_FEATURE_TOGGLE);
+
+  return walletConnectEnabled;
+}
+
+function* init() {
+  try {
+    const walletConnectEnabled = yield call(isWalletConnectEnabled);
+
+    if (!walletConnectEnabled) {
+      return;
+    }
+
+    const core = new Core({
+      projectId: WALLET_CONNECT_PROJECT_ID,
+    });
+
+    const metadata = {
+      name: 'Hathor',
+      description: 'Hathor Mobile Wallet',
+      url: 'https://hathor.network/',
+    };
+
+    const web3wallet = yield call(Web3Wallet.init, {
+      core,
+      metadata,
+    });
+
+    yield put(setWalletConnect({
+      web3wallet,
+      core,
+    }));
+
+    yield fork(setupListeners, web3wallet);
+
+    yield call(refreshActiveSessions);
```

Comment:
Maybe explain in a comment that you are not refreshing in this method, it's like the first wallet connect sessions get.

Reply:
Done in 865b81d8e4d9bf379fc13c79d36267c419b01952